--- SuiteSparse/AMD/Lib/Makefile.orig	2016-05-04 20:01:13.000000000 +0200
+++ SuiteSparse/AMD/Lib/Makefile	2023-01-11 21:43:51.689203711 +0100
@@ -52,7 +52,7 @@ amd_l_%.o: ../Source/amd_%.c $(INC)
 # Create the static library (C versions only)
 #-------------------------------------------------------------------------------
 
-static: $(AR_TARGET)
+static: $(AR_TARGET) libamdf77.a
 
 $(AR_TARGET): $(OBJ)
 	$(ARCHIVE) $@ $^
@@ -81,7 +81,7 @@ libamdf77.a: $(AMDF77)
 #-------------------------------------------------------------------------------
 
 # install AMD
-install: $(AR_TARGET) $(INSTALL_LIB)/$(SO_TARGET)
+install: $(AR_TARGET) libamdf77.a $(INSTALL_LIB)/$(SO_TARGET) $(INSTALL_LIB)/libamdf77.so.$(VERSION)
 
 $(INSTALL_LIB)/$(SO_TARGET): $(OBJ)
 	@mkdir -p $(INSTALL_LIB)
@@ -98,6 +98,18 @@ $(INSTALL_LIB)/$(SO_TARGET): $(OBJ)
 	chmod 644 $(INSTALL_DOC)/AMD_UserGuide.pdf
 	chmod 644 $(INSTALL_DOC)/AMD_README.txt
 
+$(INSTALL_LIB)/libamdf77.so.$(VERSION): $(AMD_F77)
+	@mkdir -p $(INSTALL_LIB)
+	@mkdir -p $(INSTALL_INCLUDE)
+	@mkdir -p $(INSTALL_DOC)
+	$(F77) $(SO_OPTS) $^ -o $@ -lm
+	( cd $(INSTALL_LIB) ; ln -sf libamdf77.so.$(VERSION) libamdf77.so )
+	( cd $(INSTALL_LIB) ; ln -sf libamdf77.so.$(VERSION) libamdf77.so.$(SO_VERSION) )
+	$(CP) ../Include/amd.h $(INSTALL_INCLUDE)
+	$(CP) ../Doc/AMD_UserGuide.pdf $(INSTALL_DOC)
+	$(CP) ../README.txt $(INSTALL_DOC)/AMD_README.txt
+	chmod 755 $(INSTALL_LIB)/libamdf77.so.$(VERSION)
+
 # uninstall AMD
 uninstall:
 	$(RM) $(INSTALL_LIB)/$(SO_TARGET)
