--- SuiteSparse-7.12.2/CSparse/Include/cs.h.orig	2026-02-10 19:25:51.000000000 +0100
+++ SuiteSparse-7.12.2/CSparse/Include/cs.h	2026-07-17 18:26:51.149905855 +0200
@@ -22,6 +22,11 @@
 #define csi int64_t
 #endif
 
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 void csparse_version (int version [3]) ;   // return version number
 
 /* --- primary CSparse routines and data structures ------------------------- */
@@ -151,4 +156,9 @@ csn *cs_ndone (csn *N, cs *C, void *w, v
 #define CS_MARK(w,j) { w [j] = CS_FLIP (w [j]) ; }
 #define CS_CSC(A) (A && (A->nz == -1))
 #define CS_TRIPLET(A) (A && (A->nz >= 0))
+
+#ifdef __cplusplus
+}
+#endif
+
 #endif
