--- SuiteSparse-7.3.1/AMD/CMakeLists.txt.orig	2023-11-03 23:42:14.000000000 +0100
+++ SuiteSparse-7.3.1/AMD/CMakeLists.txt	2026-07-16 19:44:10.506174634 +0200
@@ -82,10 +82,9 @@ include_directories ( Source Include )
 #-------------------------------------------------------------------------------
 
 if ( NOT NFORTRAN )
-    file ( GLOB AMD_SOURCES "Source/*.c" "Source/*.f" )
-else ( )
-    file ( GLOB AMD_SOURCES "Source/*.c" )
+    file ( GLOB AMDF77_SOURCES "Source/*.f" )
 endif ( )
+    file ( GLOB AMD_SOURCES "Source/*.c" )
 
 add_library ( AMD SHARED ${AMD_SOURCES} )
 set_target_properties ( AMD PROPERTIES
@@ -100,6 +99,17 @@ set_target_properties ( AMD PROPERTIES
 target_include_directories ( AMD
     INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/Include>
               $<INSTALL_INTERFACE:${SUITESPARSE_INCLUDEDIR}> )
+ 
+if ( NOT NFORTRAN )
+    add_library ( AMDF77 SHARED ${AMDF77_SOURCES} )
+    set_target_properties ( AMDF77 PROPERTIES
+        VERSION ${AMD_VERSION_MAJOR}.${AMD_VERSION_MINOR}.${AMD_VERSION_SUB}
+        C_STANDARD 11
+        C_STANDARD_REQUIRED ON
+        OUTPUT_NAME amdf77
+        SOVERSION ${AMD_VERSION_MAJOR}
+        WINDOWS_EXPORT_ALL_SYMBOLS ON )
+endif ( )
 
 #-------------------------------------------------------------------------------
 # static amd library properties
@@ -123,6 +133,15 @@ if ( NOT NSTATIC )
         INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/Include>
                   $<INSTALL_INTERFACE:${SUITESPARSE_INCLUDEDIR}> )
 
+    if ( NOT NFORTRAN )
+        add_library ( AMDF77_static STATIC ${AMDF77_SOURCES} )
+        set_target_properties ( AMDF77_static PROPERTIES
+            VERSION ${AMD_VERSION_MAJOR}.${AMD_VERSION_MINOR}.${AMD_VERSION_SUB}
+            C_STANDARD 11
+            C_STANDARD_REQUIRED ON
+            OUTPUT_NAME amdf77
+            SOVERSION ${AMD_VERSION_MAJOR} )
+    endif ( )
 endif ( )
 
 #-------------------------------------------------------------------------------
@@ -133,11 +152,20 @@ endif ( )
 target_link_libraries ( AMD PRIVATE SuiteSparse::SuiteSparseConfig )
 target_include_directories ( AMD PUBLIC
     "$<TARGET_PROPERTY:SuiteSparse::SuiteSparseConfig,INTERFACE_INCLUDE_DIRECTORIES>" )
+if ( NOT NFORTRAN )
+    target_link_libraries ( AMDF77 PRIVATE ${SUITESPARSE_CONFIG_LIBRARIES} )
+endif ( )
 if ( NOT NSTATIC )
     if ( TARGET SuiteSparse::SuiteSparseConfig_static )
         target_link_libraries ( AMD_static PUBLIC SuiteSparse::SuiteSparseConfig_static )
+        if ( NOT NFORTRAN )
+            target_link_libraries ( AMDF77_static PUBLIC SuiteSparse::SuiteSparseConfig_static )
+        endif ( )
     else ( )
         target_link_libraries ( AMD_static PUBLIC SuiteSparse::SuiteSparseConfig )
+        if ( NOT NFORTRAN )
+            target_link_libraries ( AMDF77_static PUBLIC SuiteSparse::SuiteSparseConfig )
+        endif ( )
     endif ( )
 endif ( )
 
@@ -162,10 +190,18 @@ install ( TARGETS AMD
     ARCHIVE DESTINATION ${SUITESPARSE_LIBDIR}
     RUNTIME DESTINATION ${SUITESPARSE_BINDIR}
     PUBLIC_HEADER DESTINATION ${SUITESPARSE_INCLUDEDIR} )
+if ( NOT NFORTRAN )
+    install ( TARGETS AMDF77
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} )
+endif ( )
 if ( NOT NSTATIC )
     install ( TARGETS AMD_static
         EXPORT AMDTargets
         ARCHIVE DESTINATION ${SUITESPARSE_LIBDIR} )
+    if ( NOT NFORTRAN )
+        install ( TARGETS AMDF77_static
+            ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} )
+    endif ( )
 endif ( )
 
 # create (temporary) export target file during build
