--- SuiteSparse-6.0.1/SPQR/CMakeLists.txt.orig	2022-11-14 01:04:25.000000000 +0100
+++ SuiteSparse-6.0.1/SPQR/CMakeLists.txt	2025-06-29 13:26:27.497135399 +0200
@@ -193,9 +193,9 @@ endif ( )
 # BLAS:
 message ( STATUS "BLAS libraries:      ${BLAS_LIBRARIES} ")
 message ( STATUS "BLAS linker flags:   ${BLAS_LINKER_FLAGS} ")
-target_link_libraries ( spqr PUBLIC ${BLAS_LIBRARIES} )
+target_link_libraries ( spqr PUBLIC ${BLAS_LIBRARIES} ${LAPACK_LIBRARIES})
 if ( NOT NSTATIC )
-target_link_libraries ( spqr_static PUBLIC ${BLAS_LIBRARIES} )
+target_link_libraries ( spqr_static PUBLIC ${BLAS_LIBRARIES} ${LAPACK_LIBRARIES})
 endif ( )
 include_directories ( ${BLAS_INCLUDE_DIRS} )
 
--- SuiteSparse-6.0.1/Mongoose/CMakeLists.txt.orig	2022-11-14 01:04:25.000000000 +0100
+++ SuiteSparse-6.0.1/Mongoose/CMakeLists.txt	2025-06-29 20:30:50.805855483 +0200
@@ -175,7 +175,7 @@ set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMA
 set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 
 # Build the Mongoose library
-add_library(mongoose_lib ${MONGOOSE_LIB_FILES})
+add_library(mongoose_lib STATIC ${MONGOOSE_LIB_FILES})
 set_property(TARGET mongoose_lib PROPERTY POSITION_INDEPENDENT_CODE ON)
 set_target_properties(mongoose_lib
         PROPERTIES OUTPUT_NAME mongoose)
