--- SuiteSparse-7.1.0/AMD/CMakeLists.txt.orig	2023-06-30 04:06:35.000000000 +0200
+++ SuiteSparse-7.1.0/AMD/CMakeLists.txt	2026-07-11 15:22:55.838103260 +0200
@@ -79,10 +79,9 @@ include_directories ( Source Include ${S
 #-------------------------------------------------------------------------------
 
 if ( NOT NFORTRAN )
-    file ( GLOB AMD_SOURCES "Source/*.c" "Source/*.f" )
-else ( )
-    file ( GLOB AMD_SOURCES "Source/*.c" )
+    file ( GLOB AMDF77_SOURCES "Source/*.c" "Source/*.f" )
 endif ( )
+    file ( GLOB AMD_SOURCES "Source/*.c" )
 
 add_library ( amd SHARED ${AMD_SOURCES} )
 set_target_properties ( amd PROPERTIES
@@ -93,6 +92,15 @@ set_target_properties ( amd PROPERTIES
     PUBLIC_HEADER "Include/amd.h"
     WINDOWS_EXPORT_ALL_SYMBOLS ON )
 
+if ( NOT NFORTRAN )
+    add_library ( amdf77 SHARED ${AMDF77_SOURCES} )
+    set_target_properties ( amdf77 PROPERTIES
+        VERSION ${AMD_VERSION_MAJOR}.${AMD_VERSION_MINOR}.${AMD_VERSION_SUB}
+        C_STANDARD 11
+        C_STANDARD_REQUIRED ON
+        SOVERSION ${AMD_VERSION_MAJOR}
+        WINDOWS_EXPORT_ALL_SYMBOLS ON )
+endif ( )
 #-------------------------------------------------------------------------------
 # static amd library properties
 #-------------------------------------------------------------------------------
@@ -110,6 +118,16 @@ if ( NOT NSTATIC )
         set_target_properties ( amd_static PROPERTIES
             OUTPUT_NAME amd_static )
     endif ( )
+
+    if ( NOT NFORTRAN )
+        add_library ( amdf77_static STATIC ${AMD_SOURCES} )
+        set_target_properties ( amdf77_static PROPERTIES
+            VERSION ${AMD_VERSION_MAJOR}.${AMD_VERSION_MINOR}.${AMD_VERSION_SUB}
+            C_STANDARD 11
+            C_STANDARD_REQUIRED ON
+            OUTPUT_NAME amdf77
+            SOVERSION ${AMD_VERSION_MAJOR} )
+    endif ( )
 endif ( )
 
 #-------------------------------------------------------------------------------
@@ -136,8 +136,14 @@ endif ( )
 
 # suitesparseconfig:
 target_link_libraries ( amd PRIVATE ${SUITESPARSE_CONFIG_LIBRARIES} )
+if ( NOT NFORTRAN )
+    target_link_libraries ( amdf77 PRIVATE ${SUITESPARSE_CONFIG_LIBRARIES} )
+endif ( )
 if ( NOT NSTATIC )
     target_link_libraries ( amd_static PUBLIC ${SUITESPARSE_CONFIG_STATIC} )
+    if ( NOT NFORTRAN )
+        target_link_libraries ( amdf77_static PRIVATE ${SUITESPARSE_CONFIG_STATIC} )
+    endif ( )
 endif ( )
 
 # libm:
@@ -142,9 +160,17 @@ install ( TARGETS amd
 install ( FILES ${CMAKE_SOURCE_DIR}/cmake_modules/FindAMD.cmake
     DESTINATION ${SUITESPARSE_LIBDIR}/cmake/SuiteSparse 
     COMPONENT Development )
+if ( NOT NFORTRAN )
+    install ( TARGETS amdf77
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} )
+endif ( )
 if ( NOT NSTATIC )
     install ( TARGETS amd_static
         ARCHIVE DESTINATION ${SUITESPARSE_LIBDIR} )
+    if ( NOT NFORTRAN )
+        install ( TARGETS amdf77_static
+            ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} )
+    endif ( )
 endif ( )
 
 #-------------------------------------------------------------------------------
