--- SuiteSparse-5.7.2/SuiteSparse_config/SuiteSparse_config.mk.orig	2020-04-08 22:42:59.000000000 +0200
+++ SuiteSparse-5.7.2/SuiteSparse_config/SuiteSparse_config.mk	2023-09-07 20:41:07.502570836 +0200
@@ -137,7 +137,7 @@ SUITESPARSE_VERSION = 5.7.2
 
     # The CF macro is used by SuiteSparse Makefiles as a combination of
     # CFLAGS, CPPFLAGS, TARGET_ARCH, and system-dependent settings.
-    CF ?= $(CFLAGS) $(CPPFLAGS) $(TARGET_ARCH) $(OPTIMIZATION) -fexceptions -fPIC
+    CF ?= $(CFLAGS) $(CPPFLAGS) $(TARGET_ARCH) -fPIC
 
     #---------------------------------------------------------------------------
     # code formatting (for Tcov on Linux only)
@@ -174,7 +174,7 @@ SUITESPARSE_VERSION = 5.7.2
         else
             # use the OpenBLAS at http://www.openblas.net (CAN BE VERY SLOW;
             # CHOLMOD can be slowed down by a factor of 100x in extreme cases)
-            BLAS ?= -lopenblas
+            BLAS ?= -lblas
             LAPACK ?= -llapack
         endif
     endif
@@ -200,7 +200,7 @@ SUITESPARSE_VERSION = 5.7.2
     # A Fortran compiler is optional.  Only required for the optional Fortran
     # interfaces to AMD and UMFPACK.  Not needed by 'make' or 'make install'
     F77 ?= gfortran
-    F77FLAGS ?= $(FFLAGS) $(OPTIMIZATION)
+    F77FLAGS ?= $(FFLAGS)
 
     #---------------------------------------------------------------------------
     # NVIDIA CUDA configuration for CHOLMOD and SPQR
@@ -358,7 +358,7 @@ SUITESPARSE_VERSION = 5.7.2
 
     ifeq ($(UNAME),Linux)
         # add the realtime library, librt, and SuiteSparse/lib
-        LDLIBS += -lrt -Wl,-rpath=$(INSTALL_LIB)
+        LDLIBS += -lrt
     endif
 
     #---------------------------------------------------------------------------
--- SuiteSparse/CSparse/Lib/Makefile.orig	2017-12-23 20:00:31.000000000 +0100
+++ SuiteSparse/CSparse/Lib/Makefile	2023-01-13 16:42:59.747528205 +0100
@@ -15,7 +15,7 @@
 # CSparse/Lib.  It does not install it for system-wide usage.
 
 LIBRARY = libcsparse
-CF = $(CFLAGS) $(CPPFLAGS) $(TARGET_ARCH) -O
+CF = $(CFLAGS) $(CPPFLAGS) $(TARGET_ARCH)
 
 I = -I../Include
 RANLIB = ranlib
--- SuiteSparse/CSparse/Demo/Makefile.orig	2017-12-23 20:05:19.000000000 +0100
+++ SuiteSparse/CSparse/Demo/Makefile	2023-01-13 16:43:15.587442393 +0100
@@ -1,4 +1,4 @@
-CF = $(CFLAGS) $(CPPFLAGS) $(TARGET_ARCH) -O
+CF = $(CFLAGS) $(CPPFLAGS) $(TARGET_ARCH)
 I = -I../Include
 
 LDLIBS += -lm
--- SuiteSparse-5.7.2/Makefile.orig	2020-04-08 22:42:59.000000000 +0200
+++ SuiteSparse-5.7.2/Makefile	2023-09-07 20:42:22.222166045 +0200
@@ -12,7 +12,6 @@ include SuiteSparse_config/SuiteSparse_c
 # Compile the default rules for each package
 go: metis
 	( cd SuiteSparse_config && $(MAKE) )
-	( cd Mongoose  && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' )
 	( cd AMD && $(MAKE) )
 	( cd BTF && $(MAKE) )
 	( cd CAMD && $(MAKE) )
@@ -30,7 +29,6 @@ ifneq ($(GPU_CONFIG),)
 	( cd GPUQREngine && $(MAKE) )
 endif
 	( cd SPQR && $(MAKE) )
-	( cd GraphBLAS && $(MAKE) JOBS=$(JOBS) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' )
 #	( cd PIRO_BAND && $(MAKE) )
 #	( cd SKYLINE_SVD && $(MAKE) )
 
@@ -38,7 +36,6 @@ endif
 # (note that CSparse is not installed; CXSparse is installed instead)
 install: metisinstall
 	( cd SuiteSparse_config && $(MAKE) install )
-	( cd Mongoose  && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' install )
 	( cd AMD && $(MAKE) install )
 	( cd BTF && $(MAKE) install )
 	( cd CAMD && $(MAKE) install )
@@ -55,7 +52,6 @@ ifneq (,$(GPU_CONFIG))
 	( cd GPUQREngine && $(MAKE) install )
 endif
 	( cd SPQR && $(MAKE) install )
-	( cd GraphBLAS && $(MAKE) JOBS=$(JOBS) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' install )
 #	( cd PIRO_BAND && $(MAKE) install )
 #	( cd SKYLINE_SVD && $(MAKE) install )
 	$(CP) README.md $(INSTALL_DOC)/SuiteSparse_README.md
@@ -116,7 +112,6 @@ endif
 # the static library
 library: metis
 	( cd SuiteSparse_config && $(MAKE) )
-	( cd Mongoose  && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' library )
 	( cd AMD && $(MAKE) library )
 	( cd BTF && $(MAKE) library )
 	( cd CAMD && $(MAKE) library )
@@ -134,7 +129,6 @@ ifneq (,$(GPU_CONFIG))
 	( cd GPUQREngine && $(MAKE) library )
 endif
 	( cd SPQR && $(MAKE) library )
-	( cd GraphBLAS && $(MAKE) JOBS=$(JOBS) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' library )
 #	( cd PIRO_BAND && $(MAKE) library )
 #	( cd SKYLINE_SVD && $(MAKE) library )
 
@@ -143,7 +137,6 @@ endif
 # both the dynamic and static libraries.
 static: metis
 	( cd SuiteSparse_config && $(MAKE) static )
-	( cd Mongoose  && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' static )
 	( cd AMD && $(MAKE) static )
 	( cd BTF && $(MAKE) static )
 	( cd CAMD && $(MAKE) static )
@@ -161,7 +154,6 @@ ifneq (,$(GPU_CONFIG))
 	( cd GPUQREngine && $(MAKE) static )
 endif
 	( cd SPQR && $(MAKE) static )
-	( cd GraphBLAS && $(MAKE) JOBS=$(JOBS) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' static )
 #	( cd PIRO_BAND && $(MAKE) static )
 #	( cd SKYLINE_SVD && $(MAKE) static )
 
@@ -219,8 +211,6 @@ clean:
 
 # Create the PDF documentation
 docs:
-	( cd GraphBLAS && $(MAKE) docs )
-	( cd Mongoose  && $(MAKE) docs )
 	( cd AMD && $(MAKE) docs )
 	( cd CAMD && $(MAKE) docs )
 	( cd KLU && $(MAKE) docs )
