--- SuiteSparse-6.0.1/SuiteSparse_config/SuiteSparse_config.h.orig	2022-11-14 01:04:25.000000000 +0100
+++ SuiteSparse-6.0.1/SuiteSparse_config/SuiteSparse_config.h	2025-06-29 11:47:15.469380302 +0200
@@ -19,11 +19,6 @@
 #ifndef SUITESPARSE_CONFIG_H
 #define SUITESPARSE_CONFIG_H
 
-#ifdef __cplusplus
-extern "C"
-{
-#endif
-
 //------------------------------------------------------------------------------
 // SuiteSparse-wide ANSI C11 #include files
 //------------------------------------------------------------------------------
@@ -89,6 +84,11 @@ extern "C"
 #include "matrix.h"
 #endif
 
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
 //------------------------------------------------------------------------------
 // string and token handling macros
 //------------------------------------------------------------------------------
--- SuiteSparse-6.0.1/AMD/Include/amd.h.orig	2025-06-29 11:48:25.092336455 +0200
+++ SuiteSparse-6.0.1/AMD/Include/amd.h	2025-06-29 11:48:27.492323453 +0200
@@ -35,13 +35,13 @@
 #ifndef AMD_H
 #define AMD_H
 
+#include "SuiteSparse_config.h"
+
 /* make it easy for C++ programs to include AMD */
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-#include "SuiteSparse_config.h"
-
 SUITESPARSE_PUBLIC int amd_order  /* returns AMD_OK, AMD_OK_BUT_JUMBLED,
                                     * AMD_INVALID, or AMD_OUT_OF_MEMORY */
 (
--- SuiteSparse-6.0.1/BTF/Include/btf.h.orig	2022-11-14 01:04:25.000000000 +0100
+++ SuiteSparse-6.0.1/BTF/Include/btf.h	2025-06-29 11:49:10.875421760 +0200
@@ -90,13 +90,13 @@
 #ifndef _BTF_H
 #define _BTF_H
 
+#include "SuiteSparse_config.h"
+
 /* make it easy for C++ programs to include BTF */
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-#include "SuiteSparse_config.h"
-
 SUITESPARSE_PUBLIC
 int32_t btf_maxtrans    /* returns # of columns matched */
 (
--- SuiteSparse-6.0.1/CAMD/Include/camd.h.orig	2022-11-14 01:04:25.000000000 +0100
+++ SuiteSparse-6.0.1/CAMD/Include/camd.h	2025-06-29 11:50:04.145133173 +0200
@@ -23,13 +23,13 @@
 #ifndef CAMD_H
 #define CAMD_H
 
+#include "SuiteSparse_config.h"
+
 /* make it easy for C++ programs to include CAMD */
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-#include "SuiteSparse_config.h"
-
 SUITESPARSE_PUBLIC int camd_order /* returns CAMD_OK, CAMD_OK_BUT_JUMBLED,
                              * CAMD_INVALID, or CAMD_OUT_OF_MEMORY */
 (
--- SuiteSparse-6.0.1/CCOLAMD/Include/ccolamd.h.orig	2022-11-14 01:04:25.000000000 +0100
+++ SuiteSparse-6.0.1/CCOLAMD/Include/ccolamd.h	2025-06-29 11:55:42.756632091 +0200
@@ -16,13 +16,13 @@
 #ifndef CCOLAMD_H
 #define CCOLAMD_H
 
+#include "SuiteSparse_config.h"
+
 /* make it easy for C++ programs to include CCOLAMD */
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-#include "SuiteSparse_config.h"
-
 /* ========================================================================== */
 /* === CCOLAMD version ====================================================== */
 /* ========================================================================== */
--- SuiteSparse-6.0.1/COLAMD/Include/colamd.h.orig	2022-11-14 01:04:25.000000000 +0100
+++ SuiteSparse-6.0.1/COLAMD/Include/colamd.h	2025-06-29 11:56:15.536454507 +0200
@@ -37,17 +37,17 @@
 #ifndef COLAMD_H
 #define COLAMD_H
 
-/* make it easy for C++ programs to include COLAMD */
-#ifdef __cplusplus
-extern "C" {
-#endif
-
 /* ========================================================================== */
 /* === Include files ======================================================== */
 /* ========================================================================== */
 
 #include "SuiteSparse_config.h"
 
+/* make it easy for C++ programs to include COLAMD */
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 /* ========================================================================== */
 /* === COLAMD version ======================================================= */
 /* ========================================================================== */
--- SuiteSparse-6.0.3/CHOLMOD/Include/cholmod.h.orig	2022-12-25 04:37:26.000000000 +0100
+++ SuiteSparse-6.0.3/CHOLMOD/Include/cholmod.h	2025-06-30 06:25:16.729303259 +0200
@@ -102,11 +102,6 @@
 #define CHOLMOD_SUB_VERSION    0
 #define CHOLMOD_SUBSUB_VERSION 2
 
-/* make it easy for C++ programs to include CHOLMOD */
-#ifdef __cplusplus
-extern "C" {
-#endif
-
 /* ========================================================================== */
 /* === Include/cholmod_io64 ================================================= */
 /* ========================================================================== */
@@ -147,6 +142,11 @@ extern "C" {
 
 #include "SuiteSparse_config.h"
 
+/* make it easy for C++ programs to include CHOLMOD */
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 /* ========================================================================== */
 /* === Include/cholmod_config.h ============================================= */
 /* ========================================================================== */
--- SuiteSparse-6.0.1/CHOLMOD/GPU/cholmod_gpu_kernels.h.orig	2022-11-14 01:04:25.000000000 +0100
+++ SuiteSparse-6.0.1/CHOLMOD/GPU/cholmod_gpu_kernels.h	2025-06-29 12:00:00.441902756 +0200
@@ -15,13 +15,13 @@
 #ifndef CHOLMODGPUKERNELS_H
 #define CHOLMODGPUKERNELS_H
 
+#include "SuiteSparse_config.h"
+
 /* make it easy for C++ programs to include CHOLMOD */
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-#include "SuiteSparse_config.h"
-
 int createMapOnDevice ( int64_t *d_Map, int64_t *d_Ls, int64_t psi, int64_t nsrow ); 
 
 int createRelativeMapOnDevice ( int64_t *d_Map, int64_t *d_Ls, 
--- SuiteSparse-6.0.1/CSparse/Include/cs.h.orig	2022-11-14 01:04:25.000000000 +0100
+++ SuiteSparse-6.0.1/CSparse/Include/cs.h	2025-06-29 12:21:36.111550178 +0200
@@ -12,6 +12,11 @@
 #ifdef MATLAB_MEX_FILE
 #include "mex.h"
 #endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 #define CS_VER 4  /* CSparse Version */
 #define CS_SUBVER 0
 #define CS_SUBSUB 0
@@ -149,4 +154,9 @@ csn *cs_ndone (csn *N, cs *C, void *w, v
 #define CS_MARK(w,j) { w [j] = CS_FLIP (w [j]) ; }
 #define CS_CSC(A) (A && (A->nz == -1))
 #define CS_TRIPLET(A) (A && (A->nz >= 0))
+
+#ifdef __cplusplus
+}
+#endif
+
 #endif
--- SuiteSparse-6.0.1/CXSparse/Include/cs.h.orig	2022-11-14 01:04:25.000000000 +0100
+++ SuiteSparse-6.0.1/CXSparse/Include/cs.h	2025-06-29 12:23:59.074109017 +0200
@@ -26,6 +26,8 @@
 #ifndef _CXS_H
 #define _CXS_H
 
+#include "SuiteSparse_config.h"
+
 #ifdef __cplusplus
 #ifndef NCOMPLEX
 #include <complex>
@@ -46,7 +48,6 @@ extern "C" {
 #define CS_COPYRIGHT "Copyright (c) Timothy A. Davis, 2006-2022"
 #define CXSPARSE
 
-#include "SuiteSparse_config.h"
 #define cs_long_t       int64_t
 #define cs_long_t_id    "%" PRId64
 #define cs_long_t_max   INT64_MAX
--- SuiteSparse-6.0.1/KLU/Include/klu.h.orig	2022-11-14 01:04:25.000000000 +0100
+++ SuiteSparse-6.0.1/KLU/Include/klu.h	2025-06-29 12:24:26.447294057 +0200
@@ -13,15 +13,15 @@
 #ifndef _KLU_H
 #define _KLU_H
 
+#include "amd.h"
+#include "colamd.h"
+#include "btf.h"
+
 /* make it easy for C++ programs to include KLU */
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-#include "amd.h"
-#include "colamd.h"
-#include "btf.h"
-
 /* -------------------------------------------------------------------------- */
 /* Symbolic object - contains the pre-ordering computed by klu_analyze */
 /* -------------------------------------------------------------------------- */
--- SuiteSparse-6.0.1/LDL/Include/ldl.h.orig	2022-11-14 01:04:25.000000000 +0100
+++ SuiteSparse-6.0.1/LDL/Include/ldl.h	2025-06-29 12:25:09.873725462 +0200
@@ -9,6 +9,10 @@
 
 #include "SuiteSparse_config.h"
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 #ifdef LDL_LONG
 #define LDL_int int64_t
 #define LDL_ID  "%" PRId64
@@ -123,3 +127,6 @@ int64_t ldl_l_valid_matrix ( int64_t n,
 #define LDL_VERSION_CODE(main,sub) ((main) * 1000 + (sub))
 #define LDL_VERSION LDL_VERSION_CODE(LDL_MAIN_VERSION,LDL_SUB_VERSION)
 
+#ifdef __cplusplus
+}
+#endif
--- SuiteSparse-6.0.1/RBio/Include/RBio.h.orig	2022-11-14 01:04:25.000000000 +0100
+++ SuiteSparse-6.0.1/RBio/Include/RBio.h	2025-06-29 12:26:00.230119325 +0200
@@ -47,12 +47,12 @@
 /* include files */
 /* -------------------------------------------------------------------------- */
 
+#include "SuiteSparse_config.h"
+
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-#include "SuiteSparse_config.h"
-
 /* -------------------------------------------------------------------------- */
 /* error codes */
 /* -------------------------------------------------------------------------- */
--- SuiteSparse-6.0.1/SPEX/Include/SPEX.h.orig	2022-11-14 01:04:25.000000000 +0100
+++ SuiteSparse-6.0.1/SPEX/Include/SPEX.h	2025-06-29 12:29:05.822447218 +0200
@@ -255,6 +255,10 @@
 #include <assert.h>
 #include "SuiteSparse_config.h"
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 //------------------------------------------------------------------------------
 // Version
 //------------------------------------------------------------------------------
@@ -952,5 +956,9 @@ SPEX_info SPEX_Left_LU_solve         //
     const SPEX_options* option
 ) ;
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif
 
--- SuiteSparse-6.0.1/SPQR/Include/SuiteSparseQR_C.h.orig	2022-11-14 01:04:25.000000000 +0100
+++ SuiteSparse-6.0.1/SPQR/Include/SuiteSparseQR_C.h	2025-06-29 12:31:45.671581241 +0200
@@ -10,6 +10,8 @@
 #ifndef SUITESPARSEQR_C_H
 #define SUITESPARSEQR_C_H
 
+#include "cholmod.h"
+
 #ifdef __cplusplus
 /* If included by a C++ program, the Complex type is std::complex<double> */
 #include <complex>
@@ -17,7 +19,6 @@
 extern "C" {
 #endif
 
-#include "cholmod.h"
 #include "SuiteSparseQR_definitions.h"
 
 #ifndef __cplusplus
--- SuiteSparse-6.0.1/SPQR/Include/SuiteSparseQR.hpp.orig	2022-11-14 01:04:25.000000000 +0100
+++ SuiteSparse-6.0.1/SPQR/Include/SuiteSparseQR.hpp	2025-06-29 12:32:21.588053332 +0200
@@ -18,11 +18,8 @@
 #include <cublas_v2.h>
 #endif
 #define SUITESPARSE_GPU_EXTERN_ON
-extern "C"
-{
 #include "SuiteSparseQR_definitions.h"
 #include "cholmod.h"
-}
 #undef SUITESPARSE_GPU_EXTERN_ON
 
 // =============================================================================
--- SuiteSparse-6.0.1/UMFPACK/Include/umfpack.h.orig	2022-11-14 01:04:25.000000000 +0100
+++ SuiteSparse-6.0.1/UMFPACK/Include/umfpack.h	2025-06-29 12:52:00.021669209 +0200
@@ -18,6 +18,13 @@
 #ifndef UMFPACK_H
 #define UMFPACK_H
 
+//------------------------------------------------------------------------------
+// include files for other packages: SuiteSparse_config and AMD
+//------------------------------------------------------------------------------
+
+#include "SuiteSparse_config.h"
+#include "amd.h"
+
 /* -------------------------------------------------------------------------- */
 /* Make it easy for C++ programs to include UMFPACK */
 /* -------------------------------------------------------------------------- */
@@ -26,13 +33,6 @@
 extern "C" {
 #endif
 
-//------------------------------------------------------------------------------
-// include files for other packages: SuiteSparse_config and AMD
-//------------------------------------------------------------------------------
-
-#include "SuiteSparse_config.h"
-#include "amd.h"
-
 /* -------------------------------------------------------------------------- */
 /* size of Info and Control arrays */
 /* -------------------------------------------------------------------------- */
